exasol/python_extension_common/deployment/language_container_validator.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+from typing import Generator
+from datetime import timedelta
+import random
+import string
+from contextlib import contextmanager
+from textwrap import dedent
+
+from tenacity import retry
+from tenacity.wait import wait_fixed
+from tenacity.stop import stop_after_delay, stop_after_attempt
+
+import pyexasol     # type: ignore
+
+_DUMMY_UDF_NAME = 'DUMMY_UDF'
+
+
+def _dress_schema(schema: str | None) -> str:
+
+    if schema:
+        return f'"{schema}".'
+    return ''
+
+
+def _create_dummy_udf(conn: pyexasol.ExaConnection, language_alias: str,
+                      schema: str | None) -> None:
+
+    sql = dedent(f"""
+    CREATE OR REPLACE {language_alias} SCALAR SCRIPT {_dress_schema(schema)}"{_DUMMY_UDF_NAME}"()
+    RETURNS DECIMAL(1, 0) AS
+
+    def run(ctx):
+        return 0
+    /
+    """)
+    conn.execute(sql)
+
+
+def _call_dummy_udf(conn: pyexasol.ExaConnection, schema: str | None) -> None:
+
+    sql = dedent(f"""
+    SELECT {_dress_schema(schema)}"{_DUMMY_UDF_NAME}"()
+    GROUP BY IPROC();
+    """)
+    result = conn.execute(sql).fetchall()
+    assert result == [(0,)]
+
+
+def _delete_dummy_udf(conn: pyexasol.ExaConnection, schema: str | None) -> None:
+
+    sql = dedent(f"""
+    DROP SCRIPT IF EXISTS {_dress_schema(schema)}"{_DUMMY_UDF_NAME};"
+    """)
+    conn.execute(sql)
+
+
+@retry(reraise=True, stop=stop_after_attempt(3))
+def _create_random_schema(conn: pyexasol.ExaConnection, schema_name_length: int) -> str:
+
+    schema = ''.join(random.choice(string.ascii_letters)
+                     for _ in range(schema_name_length))
+    sql = f'CREATE SCHEMA "{schema}";'
+    conn.execute(query=sql)
+    return schema
+
+
+def _delete_random_schema(conn: pyexasol.ExaConnection, schema: str) -> None:
```

Comment:
Is it just `_delete_schema`? It does not seem to have special handling for "randomness".

Reply:
True, I just wanted to pair up this function with the `_create_random_schema`. I was in doubt myself. OK, will remove "_random".